ovirt_imageio/client/_ui.py

```diff
@@ -28,13 +72,9 @@ def __init__(self, phase=None, error_phase="command failed", size=None,
         """
         self._phase = phase
         self._error_phase = error_phase
-        self._size = size
         self._output = output
-        # TODO: use current terminal width instead.
-        self._width = width
-        self._now = now
```

Comment:
Could be. Moved it only because it was not really used anywhere else but inside the formatter. So made not much sense for the progress bar to store this.

If I really do this, remove the draw method from the progress bar class, and pass all the arguments to the formatter, all draw calls in the ProgressBar get transformed into:

```
        self._format.draw(
            elapsed=self.elapsed(),
            size=self._size,
            value=self._value,
            transferred=self._done,
            phase=self._phase
        )
```

Which is not the nicest, either.

Reply:
Not clear what is not nice.

Reply:
It changes a plain `draw()` with that big chunk with many arguments in multiple locations, adding quite a few lines of code.

Given that the arguments are always the same, except for the `last` one, we could keep the proxy method in `ProgressBar` to call to the formatter.